src/shared/components/ResourceEditor/useEditorTooltip.tsx

```diff
@@ -0,0 +1,405 @@
+  results: TDELink[];
+  onClick: (result: TDELink) => void;
+}) {
+  const tooltipContent = document.createElement('div');
+  tooltipContent.className = clsx(
+    `${CODEMIRROR_HOVER_CLASS}-resources-content`
+  );
+  // create node for each link in results and then append it to the tooltipContent
+  (results as TDELink[]).forEach((link: TDELink) => {
+    const linkNode = createTooltipNode({
+      tag: link.resource ? `${link.resource?.[0]}/${link.resource?.[1]}` : null,
+      title: link.title ?? link._self,
+      isDownloadable: link.isDownloadable,
+    });
+    linkNode.onclick = () => {
+      removePopoversFromDOM();
+      onClick(link);
+    };
+    return tooltipContent.appendChild(linkNode);
+  });
+  return tooltipContent;
+}
+function useEditorTooltip({
+  ref,
+  isEditing,
+  orgLabel,
+  projectLabel,
+}: {
+  ref: React.MutableRefObject<CodeMirror.Editor | undefined>;
+  isEditing: boolean;
+  orgLabel: string;
+  projectLabel: string;
+}) {
+  const nexus = useNexusContext();
+  const {
+    config: { apiEndpoint },
+  } = useSelector((state: RootState) => ({
+    config: state.config,
+  }));
+
+  const allowTooltip = !isEditing;
+
+  React.useEffect(() => {
+    const currentEditor = (ref as React.MutableRefObject<CodeMirror.Editor>)
+      ?.current;
+    const editorWrapper = currentEditor.getWrapperElement();
+    function positionner(ev: MouseEvent, tooltip: HTMLDivElement) {
+      const editorRect = (ev.target as HTMLElement).getBoundingClientRect();
+      const tooltipRect = tooltip.getBoundingClientRect();
+      if (tooltipRect.height <= editorRect.top) {
+        tooltip.style.top = `${editorRect.top - tooltipRect.height}px`;
+      } else {
+        tooltip.style.top = `${editorRect.bottom}px`;
+      }
+      tooltip.style.left = `${editorRect.left}px`;
+    }
+    function removeTooltipFromDom(tooltip: HTMLDivElement) {
+      if (tooltip.parentNode) {
+        tooltip.parentNode.removeChild(tooltip);
+      }
+    }
+    function hideTooltip(tooltip: HTMLDivElement) {
+      if (!tooltip.parentNode) {
+        return;
+      }
+      setTimeout(() => {
+        removeTooltipFromDom(tooltip);
+      }, 300);
+    }
+    function showTooltip(content: HTMLDivElement, node: HTMLElement) {
+      const tooltip = document.createElement('div');
+      tooltip.className = CODEMIRROR_HOVER_CLASS;
+      tooltip.appendChild(content);
+      document.body.appendChild(tooltip);
+
+      function hide() {
+        if (tooltip) {
+          node.classList.remove('has-tooltip');
+          hideTooltip(tooltip);
+          tooltip.remove();
+        }
+        node.removeEventListener('mouseout', hide);
+        node.removeEventListener('click', hide);
+        node.removeEventListener('scroll', hide);
+      }
+
+      node.addEventListener('mouseout', hide);
+      node.addEventListener('click', hide);
+      node.addEventListener('scroll', hide);
+
+      const pool: ReturnType<typeof setTimeout> = setTimeout(() => {
```

Comment:
1. Instead of `ReturnType<typeof setTimeout>` we could just say `number` because I think it's a relatively well known fact that setTimeout returns a number. (I'd even say that a type is not strictly necessary here because it's pretty clear it's always going to be a number, that might be too opinionated though).

2. Why `pool`? (As in, it's a pool of what? It looks like it's just a single timeout, not a pool of many timeouts). What about something like `timeoutId` instead? 

3. I'm not clear about why we need this setTimeout. The `hide` function already takes care of removing the event listeners, and calls the `hideTooltip` too.

Reply:
1. The returned type of the `setTimeout` is a number but actually is a `Timer` that has many properties,
 the real type is `NodeJS.Timeout` that implement a `Timer`, I was able to make the type  `NodeJS.Timeout` but I don't confuse the developer by a NodeJS type in React. (make it `number` does't hurt also').
 
2. it's Okey to name it a pool since `setTimeout` is using a pool of ids but i will rename it for clarity

3. A tooltip is meant to be short-live component so if the browser does not capture any of the events listed in the hide  (cleanup) fn that we should destroy the DOM node